torchquad/tests/integration_test_functions.py

```diff
@@ -82,28 +92,75 @@ def get_order(self):
         """
         return inf if self.order is None else self.order
 
+    def integrand_scaling(self, integrand):
+        """_summary_
+
+        Args:
+            integrand (_type_): the integrand to be multiplied by the scaling grid
+
+        Returns:
+            Union[int, anp.ndarray]: The scaled integrand
+        """
+        integrand_scaling = self._integrand_scaling
+        if (self.is_integrand_1d):
+            return integrand_scaling * integrand
+        if (self._is_integrand_tensor):
+            scaling_einsum = "".join([chr(i + 65) for i in range(len(self.integrand_dims))])
+            return anp.einsum(f"i,{scaling_einsum}->i{scaling_einsum}", integrand, integrand_scaling)
+
+    @property
+    def _integrand_scaling(self):
+        """Get the integrand scaling factors dependent on integrand_dims
+
+        Returns:
+            Union[int, anp.ndarray]: The scaling factors
+        """
+        if (self.is_integrand_1d):
+            return 1
+        if (self._is_integrand_tensor):
+            return anp.arange(anp.prod(self.integrand_dims)).reshape(self.integrand_dims)
+        raise NotImplementedError(f"Integrand testing not implemented for dimensions {str(self.integrand_dims)}")
+
+    @property
+    def is_integrand_1d(self):
+        return self.integrand_dims == 1 or (
+                len(self.integrand_dims) == 1
+                and self.integrand_dims[0] == 1
+            )
+    
+    @property
+    def _is_integrand_tensor(self):
+        return len(self.integrand_dims) > 1 or (
+                len(self.integrand_dims) == 1
+                and self.integrand_dims[0] > 1
+            )
```

Comment:
Ditto about handling the checking for multi-dimensionality

Reply:
is there still an open question here?

Reply:
No I think this is resolved.